bindings/pyroot/pythonizations/python/ROOT/_pythonization/_uhi.py

```diff
@@ -367,42 +324,49 @@ def _slice_set(self, index, unprocessed_index, value):
     """
     import numpy as np
 
+    import ROOT
+
+    if isinstance(value, (list, range)):
+        value = np.array(value)
+
     # Depending on the shape of the array provided, we can set or not the flow bins
     # Setting with a scalar does not set the flow bins
-    include_flow_bins = not (
-        (isinstance(value, np.ndarray) and value.shape == _shape(self, include_flow_bins=False)) or np.isscalar(value)
+    include_flow_bins = (
+        (isinstance(value, np.ndarray) and value.shape != _shape(_slice_get(self, index, unprocessed_index), include_flow_bins=False)) or np.isscalar(value)
```

Comment:
This now always involves getting an histogram slice and not doing really anything with it other than getting its shape. Maybe it's necessary at the moment, I just wanted to point it out because it looks like more work than necessary.